src/distilabel/pipeline.py

```diff
@@ -522,7 +522,15 @@ def _build_dataset(  # noqa: C901
         _dataset = Dataset(
             arrow_table=dataset.flatten_indices().data, split=Split.TRAIN
         )
-        _dataset = _dataset.map(lambda _: {**generations.pop(0), **labels.pop(0)})  # type: ignore
+        # If no labeller is provided, the labels will be empty, so we create a fake dict
+        if not labels:
+            poped_label = {}
+        else:
+            poped_label = labels.pop(0)
+            # When we have futures, we cannot apply the operator **
+            if isinstance(poped_label, Future):
+                poped_label = poped_label.result()
+        _dataset = _dataset.map(lambda _: {**generations.pop(0), **poped_label})  # type: ignore
```

Comment:
Isn't the issue that in https://github.com/argilla-io/distilabel/pull/151 the `labels` variable has been renamed to `processed_labels`? IMO the following should do the work:

```suggestion
        _dataset = _dataset.map(lambda _: {**generations.pop(0), **processed_labels.pop(0)}) # type: ignore
```

Reply:
you were right, didn't noticed that!